rest/admin_api.go

```diff
@@ -678,46 +689,155 @@ func (h *handler) handleGetDbAuditConfig() error {
 	showOnlyFilterable := h.getBoolQuery("filterable")
 	verbose := h.getBoolQuery("verbose")
 
-	isEnabledFn := func(id base.AuditID) bool {
-		_, ok := h.db.Options.LoggingConfig.Audit.EnabledEvents[id]
-		return ok
+	var (
+		etagVersion    string
+		dbAuditEnabled bool
+		enabledEvents  = make(map[base.AuditID]struct{})
+	)
+
+	if h.server.BootstrapContext.Connection != nil {
+		found, dbConfig, err := h.server.fetchDatabase(h.ctx(), h.db.Name)
+		if err != nil {
+			return err
+		}
+
+		if !found {
+			return base.HTTPErrorf(http.StatusNotFound, "database config not found")
+		}
+
+		etagVersion = dbConfig.Version
+
+		runtimeConfig, err := MergeDatabaseConfigWithDefaults(h.server.Config, &dbConfig.DbConfig)
+		if err != nil {
+			return err
+		}
+
+		// grab runtime version of config, so that we can see what events would be enabled
+		if runtimeConfig.Logging != nil && runtimeConfig.Logging.Audit != nil {
+			dbAuditEnabled = base.BoolDefault(runtimeConfig.Logging.Audit.Enabled, false)
+			for _, event := range runtimeConfig.Logging.Audit.EnabledEvents {
+				enabledEvents[base.AuditID(event)] = struct{}{}
+			}
+		}
 	}
 
-	// TODO: Move to structs
-	events := make(map[string]interface{}, len(base.AuditEvents))
+	events := make(map[string]any, len(base.AuditEvents))
 	for id, descriptor := range base.AuditEvents {
 		if showOnlyFilterable && !descriptor.FilteringPermitted {
 			continue
 		}
+
 		idStr := id.String()
+		_, eventEnabled := enabledEvents[id]
+
 		if verbose {
-			events[idStr] = map[string]interface{}{
-				"name":        descriptor.Name,
-				"description": descriptor.Description,
-				"enabled":     isEnabledFn(id),
-				"filterable":  descriptor.FilteringPermitted,
+			events[idStr] = handleDbAuditConfigBodyVerboseEvent{
+				Name:        stringPtrOrNil(descriptor.Name),
+				Description: stringPtrOrNil(descriptor.Description),
+				Enabled:     &eventEnabled,
+				Filterable:  base.BoolPtr(descriptor.FilteringPermitted),
 			}
 		} else {
-			events[idStr] = isEnabledFn(id)
+			events[idStr] = &eventEnabled
 		}
 	}
 
-	h.writeJSON(events)
+	resp := handleDbAuditConfigBody{
+		Enabled: &dbAuditEnabled,
+		Events:  events,
+	}
 
+	h.setEtag(etagVersion)
+	h.writeJSON(resp)
 	return nil
 }
 
 // PUT/POST audit config for database
 func (h *handler) handlePutDbAuditConfig() error {
```

Comment:
I think this should still have `h.assertAdminOnly()`

Reply:
`mutateDbConfig` does that